vec/src/atomic.rs

```diff
@@ -0,0 +1,35 @@
+/// Atomic 
+///
+/// Intended to be a generic trait for all vector-able elements. Ideally
+/// would encompass things like coercion traits and operator implementations.
+/// For now just requires Clone.
+/// 
+use super::coercion::*;
+pub trait Atomic: 
+    Clone + 
+    NaAble + 
+    IntoLogical + 
+    IntoNumeric + 
+    AtomicMode {}
+
+pub trait AtomicMode {
+    fn is_numeric() -> bool { false }
+    fn is_logical() -> bool { false }
+    fn is_integer() -> bool { false }
+    fn is_character() -> bool { false }
+}
```

Comment:
In this rework, internal types are just a type constraint! I added the mode for quickly checking the type, but from R's perspective these are just _Vectors_, and casting between types for logical operations or math all happen in the background. :tada: